src/test/java/io/securecodebox/persistence/defectdojo/service/UserProfileServiceTest.java

```diff
@@ -0,0 +1,59 @@
+
+package io.securecodebox.persistence.defectdojo.service;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import io.securecodebox.persistence.defectdojo.config.DefectDojoConfig;
+import java.net.URISyntaxException;
+import static org.junit.jupiter.api.Assertions.assertIterableEquals;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.client.MockRestServiceServer;
+import static org.springframework.test.web.client.match.MockRestRequestMatchers.requestTo;
+import static org.springframework.test.web.client.response.MockRestResponseCreators.withSuccess;
+
+
+// This test is special because the defectdojo api does not return a list, 
+// but the generic code assumes every endpoint returns a list
+public class UserProfileServiceTest {
+    
+    DefectDojoConfig config;
+    UserProfileService underTest;
+    MockRestServiceServer mockServer;
+    
+    // This string does not contain every field of the api response as those are not implemented
+    String apiResponse = """
+                         {
+                           "user": {
+                             "id": 0,
+                             "username": "GdqmXprK.j7R+OYE49SzL3mM2U6I0DyLRHnDg87i9It0AfP-kxvswW3qOI2i+31-@0",
+                             "first_name": "string",
+                             "last_name": "string",
+                             "email": "user@example.com",
+                             "last_login": "2022-11-01T16:20:19.373Z",
+                             "is_active": true,
+                             "is_superuser": true,
+                             "configuration_permissions": [0]
+                           }
+                         }
+                         """;
+    
+    @BeforeEach
+    void setup() {
+        config = new DefectDojoConfig("https://defectdojo.example.com", "abc", "test-user", 42);
+        underTest = new UserProfileService(config);
+        mockServer = MockRestServiceServer.createServer(underTest.getRestTemplate());
+    }
+    
+    @Test
+    void testSearch() throws JsonProcessingException, URISyntaxException {
+        var url = config.getUrl() + "/api/v2/" + underTest.getUrlPath() + "/?offset=0&limit=100";        
+        mockServer.expect(requestTo(url)).andRespond(withSuccess(apiResponse, MediaType.APPLICATION_JSON));
+        
+        var expected = underTest.deserializeList(apiResponse).getResults();
```

Comment:
I don't think it is a good idea to generate the expected result form the `underTest` class.
If the implementation is broken it would probably also be broken here and we wouldn't know about it as the expected and actual are the same thing.